### PR TITLE
Small refactor in provider

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR:

Resolves: #<Issue number>

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/eth_provider/provider.rs [hunk 1]
```diff
@@ -625,7 +620,7 @@ where
         &self,
         block_id: Option<BlockId>,
     ) -> EthProviderResult<Option<Vec<reth_rpc_types::Transaction>>> {
-        let block_id = match block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest)) {
+        let block_id = match block_id.unwrap_or_default() {
```
Comment (greged93) [inline src/eth_provider/provider.rs:623]: I don't know if I am a fan of that, I know  ok, but performance wise it won't change anything and this makes it more readable, you know straight what the default block id is

File: src/eth_rpc/servers/alchemy_rpc.rs [hunk 2]
```diff
@@ -23,11 +23,9 @@ impl<P: EthereumProvider> AlchemyRpc<P> {
 impl<P: EthereumProvider + Send + Sync + 'static> AlchemyApiServer for AlchemyRpc<P> {
     #[tracing::instrument(skip_all, ret, fields(address = %address, token_addresses = ?token_addresses))]
     async fn token_balances(&self, address: Address, token_addresses: Vec<Address>) -> Result<TokenBalances> {
-        let block_id = BlockId::Number(BlockNumberOrTag::Latest);
-
         let handles = token_addresses.into_iter().map(|token_addr| {
             let token = EthereumErc20::new(token_addr, &self.eth_provider);
-            let balance = token.balance_of(address, block_id);
+            let balance = token.balance_of(address, Default::default());
```
Comment (greged93) [inline src/eth_rpc/servers/alchemy_rpc.rs:28]: same comment here

File: src/eth_provider/provider.rs [hunk 3]
```diff
@@ -595,12 +595,10 @@ where
     }
 
     async fn block_receipts(&self, block_id: Option<BlockId>) -> EthProviderResult<Option<Vec<TransactionReceipt>>> {
-        let block_id = block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest));
-        match block_id {
+        match block_id.unwrap_or_default() {
```
Comment (tcoratger) [inline src/eth_provider/provider.rs:598]:
```suggestion
        match block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest)) {
```

File: src/eth_provider/provider.rs (hunk 1 above)
Comment (tcoratger) [inline src/eth_provider/provider.rs:623]:
```suggestion
        let block_id = match block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest)) {
```

Comment (tcoratger) [reply src/eth_provider/provider.rs]: Yes understood, this is right that for readability, old solution was better. This was to lighten the code, but let's stay like that, just reversed :)

**Review by greged93 — APPROVED** [review]

**Pull request merged**